test/src/l2_test.cpp

```diff
@@ -170,83 +103,12 @@ TEST_F(L2, link_output) {
   bm_free(buf);
 }
 
-/*!
- @brief Get number of ports and devices configured for L2
-*/
-TEST_F(L2, num_devices_ports) {
-  EXPECT_EQ(bm_l2_get_num_ports(), num_port);
-  EXPECT_EQ(bm_l2_get_num_devices(), num_devices);
-}
-
-/*!
- @brief Get the device handle of all devices configured for L2
-*/
-TEST_F(L2, get_device_handle) {
-  void *handle = NULL;
-  uint32_t start_port_idx = 0;
-  for (uint8_t i = 0; i < num_devices; i++) {
-    EXPECT_EQ(bm_l2_get_device_handle(i, &handle, &start_port_idx), true);
-    EXPECT_EQ(handle, META[i].device_handle);
-    EXPECT_EQ(start_port_idx, i * (port_per_device));
-  }
-
-  // Test invalid use cases
-  EXPECT_EQ(bm_l2_get_device_handle(0, NULL, NULL), false);
-  EXPECT_EQ(bm_l2_get_device_handle(0, &handle, NULL), false);
-  EXPECT_EQ(bm_l2_get_device_handle(0, NULL, &start_port_idx), false);
-  EXPECT_EQ(bm_l2_get_device_handle(UINT8_MAX, &handle, &start_port_idx),
-            false);
-}
-
 /*!
   @brief Test setting the power (on/off) of the all devices
 */
 TEST_F(L2, set_power) {
-  for (uint8_t i = 0; i < num_devices; i++) {
-    EXPECT_EQ(bm_l2_netif_set_power(META[i].device_handle, true), BmOK);
-    EXPECT_EQ(bm_l2_netif_set_power(META[i].device_handle, false), BmOK);
-  }
-
-  // Test invalid use cases
-  EXPECT_NE(bm_l2_netif_set_power(NULL, true), BmOK);
-}
-
-/*!
- @brief Test the callbacks initialized by the L2 interface
-*/
-TEST_F(L2, test_callbacks) {
```

Comment:
These didn't make sense anymore, but we may want to write some unit tests for the receive callback after investigating whether we need the queue and task.